frontend/src/app/workflows/workflows.component.ts

```diff
@@ -0,0 +1,194 @@
+import {HttpClient} from '@angular/common/http';
+import {Component, NgZone, OnInit, ViewChild} from '@angular/core';
+
+import Ajv from 'ajv';
+import * as Yaml from 'js-yaml';
+import {ToastService} from '../toast.service';
+import {PrivacyService} from '../privacy.service';
+import {WorkflowsService} from './workflows.service';
+import {GenerateResult} from './model/GenerateResult';
+import {IOutputData, SplitComponent} from 'angular-split';
+import {workflowsSchema} from './model/helper/workflows.schema';
+
+@Component({
+  selector: 'app-workflows',
+  templateUrl: './workflows.component.html',
+  styleUrls: ['./workflows.component.scss']
+})
+export class WorkflowsComponent implements OnInit {
+  @ViewChild('split') split!: SplitComponent;
+
+  public content!: any;
+  public codemirrorOptions: any;
+
+  public generateResult!: GenerateResult;
+
+  public currentExampleDesc: string = 'Select example';
+  public examplesList = ['New Workflow', 'Data Modelling', 'Microservices', 'Pages'];
+
+  public showIframeHider = false;
+  public newPageIndex!: number;
+  public currentDisplay: 'pages' | 'objects' | 'class' = 'pages';
+
+  private ajv!: Ajv;
+  private validate!: any;
+  public loading: boolean = false;
+
+  constructor(
+    private toastService: ToastService,
+    private zone: NgZone,
+    private fulibWorkflowsService: WorkflowsService,
+    private privacyService: PrivacyService,
+    private http: HttpClient,
+  ) {
+    // https://angular.io/api/core/NgZone
+    const generateHandler = () => this.zone.run(() => this.generate());
+
+    this.codemirrorOptions = {
+      lineNumbers: true,
+      mode: 'yaml',
+      extraKeys: {
+        'Ctrl-Space': 'autocomplete',
+        'Ctrl-S': generateHandler,
```

Comment:
I already mentioned this in voice chat, but documenting it here: This can be accomplished with the `(save)` event of the autotheme-codemirror component.

Reply:
You sure? 
The AutothemeCodemirror doesn't have a save EventEmitter. Am i simply not getting what you are trying to say? :D

Reply:
Ah I was confused, actually the scenario-codemirror component has that. But it cannot be configured so it's probably not useful for you. For now you can just keep it this way but if you ever want autosave I can refactor it into autotheme-codemirror.

Reply:
Ah I was confused, actually the scenario-codemirror component has that. But it cannot be configured so it's probably not useful for you. For now you can just keep it this way but if you ever want autosave I can refactor it into autotheme-codemirror.